internal/main

```diff
@@ -367,17 +380,59 @@ function setup {
 function setup_jenkins {
     echo "Setting up Jenkins build environment"
     echo ""
+    # One-time install of pyenv. It installs to Jenkins home dir, so it can be a shared resource.
+    [ -d "$PYENV_ROOT" ] || {
+        mkdir -p "$PYENV_ROOT"
+        curl -L https://raw.githubusercontent.com/yyuu/pyenv-installer/master/bin/pyenv-installer | bash
+        mkdir -p "$PYENV_ROOT/rax-docs-installed"
+    }
+    export PATH="$PYENV_ROOT/bin:$PATH"
+    eval "$(pyenv init -)" || fail "Installing pyenv seems to have failed. Investigate Jenkins."
+
+    # Python install. We signal to ourselves that an install is done by touching a file. If
+    # the install dir exists but the done file doesn't, we need to wait to see if someone
+    # else is building it at the same time. This is the cost of sharing installations. The
+    # time saved is worth it.
+
+    INSTALL_FILE="$PYENV_ROOT/rax-docs-installed/$PYVERSION"
+    if [ ! -d "$PYENV_ROOT/versions/$PYVERSION" ]; then
+        # Eagerly create the dir to minimize race conditions on this directory test
+        mkdir -p "$PYENV_ROOT/versions/$PYVERSION"
+        SPACE_LEFT=$(df -k --output=avail . | tail -1)
+        [ "$SPACE_LEFT" -gt $((500*1024)) ] || {
+            rmdir "$PYENV_ROOT/versions/$PYVERSION"
+            echo "Less than 500MB remaining in Jenkins home partition."
+            echo "Clean up some space before installing more Pythons."
+            exit 1
+        }
+        # Flags suggested for installing 3.4.3 on CentOS 6 to get past SSL errors
```

Comment:
Are these flags required for version > 3.4.3? Because it looks like you're installing 3.6.13.

Reply:
Again talked to @zzantozz personally, needed for 3.4.3, but he didn't bother checking if they were needed for newer versions,